### PR TITLE
Implement similar changes that @cbeer made in https://github.com/sul-…

### PR DESCRIPTION
**Pull request by mejackreed**
…dlss/sul_styles/pull/35/files to support a default sprockets approach that will still support webpacker

This only changes the test app generated.

### CODE REVIEW

File: spec/test_app_templates/lib/generators/test_app_generator.rb [hunk 1]
```diff
@@ -8,6 +8,10 @@ def remove_index
     remove_file "public/index.html"
   end
 
+  def appease_webpacker
```
Comment (jcoyne) [inline spec/test_app_templates/lib/generators/test_app_generator.rb]: Can you add a comment about why this needs to be here?

**Comment by jrochkind** [comment]
Can you leave a comment, perhaps in source, but at least here, about why `empty_directory 'app/assets/images'` is necessary to appease webpacker? What difference does it make? And why are there things in app/assets/images to begin with, and why do they need to be deleted?

When trying to debug engine_cart problems, which I've my fair share of, I find it enormously helpful to have 'breadcrumbs' left explaining why non-obvious things were done.

File: spec/test_app_templates/lib/generators/test_app_generator.rb (hunk 1 above)
Comment (mejackreed) [reply spec/test_app_templates/lib/generators/test_app_generator.rb]: Great idea. I've added in a comment. Thanks @jcoyne @jrochkind

Comment (cdmo) [reply spec/test_app_templates/lib/generators/test_app_generator.rb]: I thought sprockets looks for this to be present, not webpacker 🤔

Comment (cdmo) [reply spec/test_app_templates/lib/generators/test_app_generator.rb]: assuming (~perhaps~ likely incorrectly) that this is a fix for https://github.com/projectblacklight/blacklight/issues/2231#issuecomment-577372873

**Comment by cdmo** [comment]
I ran rake locally and all the feature specs failed with:

```
  70) Search Results has for an empty query
      Failure/Error: <%= javascript_include_tag "application" %>

      ActionView::Template::Error:
        Asset `application.js` was not declared to be precompiled in production.
        Declare links to your assets in `app/assets/config/manifest.js`.

          //= link application.js
        and restart your server
      # ./app/views/layouts/blacklight/base.html.erb:15:in `block in ___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_base_html_erb__3329786505906691760_70199829967220'
      # ./app/views/layouts/blacklight/base.html.erb:2:in `___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_base_html_erb__3329786505906691760_70199829967220'
      # ./app/views/layouts/blacklight.html.erb:17:in `___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_html_erb__2742037008966264949_70199877634540'
      # ./spec/features/search_results_spec.rb:62:in `search_for'
      # ./spec/features/search_results_spec.rb:5:in `block (2 levels) in <top (required)>'
      # ------------------
      # --- Caused by: ---
      # Sprockets::Rails::Helper::AssetNotPrecompiled:
      #   Asset `application.js` was not declared to be precompiled in production.
      #   Declare links to your assets in `app/assets/config/manifest.js`.
      #
      #     //= link application.js
      #   and restart your server
      #   ./app/views/layouts/blacklight/base.html.erb:15:in `block in ___sers_cdm____esktop_blacklight_app_views_layouts_blacklight_base_html_erb__3329786505906691760_70199829967220'
```

Is `rake` passing specs for everyone else locally on this branch?

**Comment by mjgiarlo** [comment]
@cdmo :speech_balloon: 
> I ran rake locally and all the feature specs failed with:
>
> ...

I haven't run it locally w/ this branch, but a hunch: did you regenerate the EngineCart test app after pulling this branch? If so, can you say what Ruby and Rails versions you tested with? (I'm guessing whatever the defaults are when running `rake` but thought it worth asking.)

**Comment by cdmo** [comment]
Started with a completely fresh clone, checked out this branch, ran `docker-compose up solr`, waited til that was initialized, separate tab ran `bundle exec rake`. Got 70 failing specs again, all with the same report I got above (the one that says `# --- Caused by: ---
      # Sprockets::Rails::Helper::AssetNotPrecompiled`) Maybe just me, I dont know.

**Comment by mjgiarlo** [comment]
> Started with a completely fresh clone, checked out this branch, ran `docker-compose up solr`, waited til that was initialized, separate tab ran `bundle exec rake`. Got 70 failing specs again, all with the same report I got above (the one that says `# --- Caused by: --- # Sprockets::Rails::Helper::AssetNotPrecompiled`) Maybe just me, I dont know.
> 
> EDIT: looks like `2.6.5`, going by the `DEPRECATED` notices in the log, like:
> `[DEPRECATED] Bundler.clean_system has been deprecated in favor of Bundler.unbundled_system. If you instead want to run the command in the environment before bundler was originally loaded, use Bundler.original_system (called at /Users/cdm32/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/engine_cart-2.3.0/lib/engine_cart/tasks/engine_cart.rake:115)`
> 
> [full log](https://github.com/projectblacklight/blacklight/files/4370706/rake_log.log)

I'll give this a spin today, @cdmo, once I'm done responding to reviews of #2263 

Thanks!

**Comment by mjgiarlo** [comment]
@cdmo @mejackreed I was able to pull this branch and see all specs pass. Here's what I did:

```shell
# start off clean
$ rm -rf .internal_test_app Gemfile.lock 
# specify Rails version
$ export RAILS_VERSION=6.0.2.1
# tell EngineCart to skip webpack
$ export ENGINE_CART_RAILS_OPTIONS="--skip-webpack-install"
# install dependencies
$ bundle 
# spin up solr in background
$ docker-compose up -d solr
# run default task (build app, run specs)
$ rake
```

Worked like a charm.

**Comment by mjgiarlo** [comment]
@cdmo can you give :point_up: a run-through and confirm it works and is sensible? And then, assuming so, mind re-reviewing? Want to make sure we have :100: approvals before merging.

**Comment by mjgiarlo** [comment]
@cdmo Amending what I wrote above: these steps worked because the Rails server did not get spun up. Running `rake blacklight:server` instead of just `rake` shows that the changes in this branch don't yet :100: fix the problem. @mejackreed and I are pairing now to see if we can make improvements.

**Comment by mejackreed** [comment]
Closing this for now. I think we have some other options to resolve Blacklight / Rails 6 / Webpacker

**Pull request closed without merging**